src/authorization.js

```diff
@@ -525,13 +525,14 @@ class Authorization {
   }
 
   /**
-   * Sets the agent WebID for this authorization. Implemented as `setAgent()`
-   * setter method to enforce mutual exclusivity with `group` property, until
-   * ES6 setter methods become available.
+   * Sets the agent WebID for this authorization.
    * @method setAgent
-   * @param agent {String|Quad} Agent URL (or `acl:agent` RDF triple).
+   * @param agent {string|Quad|GroupListing} Agent URL (or `acl:agent` RDF triple).
    */
   setAgent (agent) {
```

Comment:
However if you wish to stick with this approach I'd like to see the agent be abstracted away from the concerns of this method. Could we define an `Agent` class/superclass which represents either a group, individual agent, or "everyone"? That way we'd only need a single `setAgent` method, and the concern of whether the agent is an individual, group, or everyone could be handled during serialization using a duck typing approach.